test/integration/volume/attach_detach_test.go

```diff
@@ -0,0 +1,211 @@
+		},
+	}
+	return fakePod
+}
+
+func TestPodDeletionWithDswp(t *testing.T) {
+	_, s := framework.RunAMaster(nil)
+	defer s.Close()
+	namespaceName := "test-pod-deletion"
+
+	node := &api.Node{
+		ObjectMeta: api.ObjectMeta{
+			Name: "node-sandbox",
+			Annotations: map[string]string{
+				volumehelper.ControllerManagedAttachAnnotation: "true",
+			},
+		},
+	}
+
+	ns := framework.CreateTestingNamespace(namespaceName, s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+
+	testClient, ctrl, podInformer, nodeInformer := createAdClients(ns, t, s, defaultSyncPeriod)
+
+	pod := fakePodWithVol(namespaceName)
+	podStopCh := make(chan struct{})
+
+	if _, err := testClient.Core().Nodes().Create(node); err != nil {
+		t.Fatalf("Failed to created node : %v", err)
+	}
+
+	go nodeInformer.Run(podStopCh)
+
+	if _, err := testClient.Core().Pods(ns.Name).Create(pod); err != nil {
+		t.Errorf("Failed to create pod : %v", err)
+	}
+
+	go podInformer.Run(podStopCh)
+
+	// start controller loop
+	stopCh := make(chan struct{})
+	go func() {
+		ctrl.Run(stopCh)
+	}()
+
+	waitToObservePods(t, podInformer, 1)
+	podKey, err := cache.MetaNamespaceKeyFunc(pod)
+	if err != nil {
+		t.Fatalf("MetaNamespaceKeyFunc failed with : %v", err)
+	}
+
+	podInformerObj, _, err := podInformer.GetStore().GetByKey(podKey)
+
+	if err != nil {
+		t.Fatalf("Pod not found in Pod Informer cache : %v", err)
+	}
+
+	podsToAdd := ctrl.GetDesiredStateOfWorld().GetPodToAdd()
+
+	if len(podsToAdd) == 0 {
+		t.Fatalf("Pod not added to desired state of world")
+	}
+
+	// let's stop pod events from getting triggered
+	close(podStopCh)
+	err = podInformer.GetStore().Delete(podInformerObj)
+	if err != nil {
+		t.Fatalf("Error deleting pod : %v", err)
+	}
+
+	waitToObservePods(t, podInformer, 0)
+
+	// the populator loop turns every 1 minute
+	time.Sleep(2 * time.Minute)
+	podsToAdd = ctrl.GetDesiredStateOfWorld().GetPodToAdd()
+
+	if len(podsToAdd) != 0 {
+		t.Fatalf("All pods should have been removed")
+	}
+
+	close(stopCh)
+}
+
+// wait for the podInformer to observe the pods. Call this function before
+// running the RC manager to prevent the rc manager from creating new pods
+// rather than adopting the existing ones.
+func waitToObservePods(t *testing.T, podInformer cache.SharedIndexInformer, podNum int) {
+	if err := wait.Poll(10*time.Second, 60*time.Second, func() (bool, error) {
+		objects := podInformer.GetIndexer().List()
+		if len(objects) == podNum {
+			return true, nil
+		} else {
+			return false, nil
+		}
+	}); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func createAdClients(ns *api.Namespace, t *testing.T, s *httptest.Server, syncPeriod time.Duration) (*internalclientset.Clientset, attachdetach.AttachDetachController, cache.SharedIndexInformer, cache.SharedIndexInformer) {
+	config := restclient.Config{
+		Host:          s.URL,
+		ContentConfig: restclient.ContentConfig{GroupVersion: &registered.GroupOrDie(api.GroupName).GroupVersion},
+		QPS:           1000000,
```

Comment:
I simply used from other tests that do similar thing. I think overall idea is to keep it high enough so as we are not throttling API requests.